Require Linux or macOS as host (or Ubuntu via VirtualBox on Windows) (postpone MBVM-53)

yvanzo opened the pull request:
Rather than trying to support a platform we don’t run:
* Add “Linux or macOS” to requirements
* Explain how to install GNU Bash 4 on macOS
* Recommend using Ubuntu via VirtualBox on Windows
* Refactor admin helper scripts to error on unknown platform, but still allow to get around by setting `DOCKER_CMD` and `DOCKER_COMPOSE_CMD` variables in environment
* Leave MBVM-53 open for people interesting in adding instructions for Windows

----

<del>
## Related issues

* https://github.com/metabrainz/musicbrainz-docker/issues/128
* https://github.com/metabrainz/musicbrainz-docker/issues/129

## Related discussions

* https://chatlogs.metabrainz.org/brainzbot/metabrainz/2020-04-23/?msg=4571795&page=1

## To-do list

* [X] Detect Git Bash for Windows: fixed in https://github.com/metabrainz/musicbrainz-docker/commit/c2ea55bc4e73db84e37e74e626f111ecd19e5d6f
* [ ] Solve further issues…
* [ ] Update requirements for a working setup on Windows
      References:
  * https://blog.jayway.com/2017/04/19/running-docker-on-bash-on-windows/
  * https://forums.docker.com/t/weird-error-under-git-bash-msys-solved/9210
  * https://github.com/lfurzewaddock/wsldocker
  * https://www.howtogeek.com/249966/how-to-install-and-use-the-linux-bash-shell-on-windows-10/
  * https://pythonise.com/series/windows-subsystem-for-linux/docker-on-windows-subsystem-for-linux
* [ ] Update instructions to work with that setup too

yvanzo committed:
Detect Git for Windows to set docker command

SuperSandro2000 commented:
Be careful with Postgres and Windows. The both don't get along that nicely. https://github.com/docker/for-win/issues/445

yvanzo committed:
Require Linux or macOS as host (postpone MBVM-53)

A lot of setup is required to make it work with Docker for Windows
(either Docker Desktop or Docker Toolbox) and Git Bash for Windows.

yvanzo committed:
Improve required software section (with homebrew)

yvanzo commented:
In our case, this is not an issue for Postgres as the volume [`pgdata`](https://github.com/metabrainz/musicbrainz-docker/blob/mbvm-38-dev/docker-compose.yml#L8-L9) is not mounted from the host.

But a related issue occurs with indexer configuration file, see https://github.com/metabrainz/musicbrainz-docker/issues/129#issuecomment-615937331.

Anyway, even though Windows compatibility would be welcome, only people running it can actually test the compatibility. I don’t, thus I gave up working on this issue, see https://github.com/metabrainz/musicbrainz-docker/issues/129#issuecomment-618356821. However, maybe someone will be interested in and even capable of, so just documenting this with https://github.com/metabrainz/musicbrainz-docker/pull/133/commits/b72836549cdfd73502484dff04e857fa1493590e and MBVM-53 instead. I will also work on reducing the need for things that make this project lesser portable.
